exasol/toolbox/tools/security.py

```diff
@@ -0,0 +1,266 @@
+"""This module contains security related CLI tools and code"""
+import json
+import re
+import subprocess
+import sys
+from dataclasses import (
+    asdict,
+    dataclass,
+)
+from enum import Enum
+from functools import partial
+from inspect import cleandoc
+from typing import (
+    Generator,
+    Iterable,
+    Tuple,
+)
+
+import typer
+
+stdout = print
+stderr = partial(print, file=sys.stderr)
+
+
+# Note: In the long term we may want to adapt the official CVE json schema
+# https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/CVE_JSON_5.0_schema.json
+@dataclass(frozen=True)
+class Issue:
+    cve: str
+    cwe: str
+    description: str
+    coordinates: str
+    references: tuple
+
+
+def _issues(input) -> Generator[Issue, None, None]:
+    lines = (l for l in input if l.strip() != "")
+    for line in lines:
+        obj = json.loads(line)
+        yield Issue(**obj)
+
+
+def _issues_as_json_str(issues):
+    for issue in issues:
+        issue = asdict(issue)  # type: ignore
+        yield json.dumps(issue)
+
+
+def gh_security_issues() -> Generator[Tuple[str, str], None, None]:
+    """
+    Yields issue-id, cve-id pairs for all (closed, open) issues associated with CVEs
+
+    Return:
+        A generator which yield tuples of (id,title).
+
+    Raises:
+        subprocess.CalledProcessError: If the underlying command fails.
+    """
+    command = [
+        "gh",
```

Comment:
is this guaranteed to be available in the github runners?

Reply:
maybe it is better to use the API here, but this could be done in another ticket

Reply:
@tkilias my thinking was the following: Using the API directly will add an additional dependency and significant complexity. If we add more `GitHub` specific code in the 
future we can/should migrate to using the API directly.